marienfressinaud/FreshRSS
# Add auto-registration when using http_auth

Changes proposed in this pull request:

When using `http_auth`, the app trusts the web server to perform the authentication. Thus, it should also trust the web server to only pass valid users. Currently, users must be manually created by the admin before they can log in. This PR adds an option so that, when a non-existing user authenticate using `http_auth`, the user is automatically created and logged in.

How to test the feature manually:

1. Use a working `http_auth` based setup
2. Set `http_auth_auto_register` to `true` in your config file
3. Log in to a non-existing FreshRSS user
4. The user should be created and logged in, instead of showing a permission error

Pull request checklist:
